### DISCUSSION TITLE
Whether a message can be withdrawn？

### DISCUSSION THREAD

**Discussion by wjwzero** [emqx/emqx#6530] [Q&A]
Whether an EMQX offline message can be withdrawn？

**Comment by k32** [comment]
Could you please clarify what you do mean by offline message?

**Comment by wjwzero** [comment]
Usually an MQTT client receives messages only when it is connected to an EMQ X, if this client is off-line, it will not receive messages. But if a client has a fixed ClientID, and it connects to the broker with clean_seesion = false, the broker will store particular message for it when it is off-line, if the Pub/Sub is done at certain QoS level (broker configuration). These messages will be delivered when this client is reconnected.

A requirement ,withdrawn this off-lien message,if it is a wrong message.

**Reply by zmstone** [reply]
Hi @wjwzero 

The queued messages are opaque and they are stored in a “queue” which has not apis for inspection ( to check if it’s “wrong”) or update (to “withdraw”)
